orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -411,7 +371,7 @@ def forward(ctx, *inputs, **kwargs):
                 # Run and return module outputs.
                 user_outputs = tuple(_ort_output_to_torch_tensor(forward_output) \
                     for forward_output in self._training_session.run_forward(self._training_io_binding, self._run_options))
-                return user_outputs[0] if len(user_outputs) == 1 else user_outputs
+                return user_outputs
```

Comment:
Will outputs with len 1 be returned as a list with single element or just the element directly?

Reply:
The element will be returned directly. This output is not the output the users would see. This is the output from the autograd forward method. This output then gets processed by the output defined inside ```ORTMOdule``` which expects that the output_names sequence have the same length as the number of outputs.
Which is why the entire sequence needs to be returned here before it gets processed by ```ORTModule``` ```forward``` method.